datafusion/expr/src/udaf.rs

```diff
@@ -196,6 +196,10 @@ impl AggregateUDF {
         self.inner.state_fields(args)
     }
 
+    pub fn fields(&self, args: StateFieldsArgs) -> Result<Field> {
```

Comment:
Could we document this function and what it is for (also in AggregateUdfImpl)? 

Also, the name is strange to me -- it is `fields` but it returns a single `Field` and the corresponding function on `AggregateUDFImpl` is called `field` (no `s`) 🤔